### DISCUSSION TITLE
docsify with web-components?

### DISCUSSION THREAD

**Discussion by ghost** [docsifyjs/docsify#1940] [Q&A]
Hi everyone.

#### Idea
I would like to know if you have already thought about making docsify in a web component. You all have already thought about making docsify in a web component?

#### what is web component?
Web Components is a suite of different technologies allowing you to create reusable custom elements — with their functionality encapsulated away from the rest of your code — and utilize them in your web apps.

#### my idea and concept
```html
<!doctype html>
<html>
<head>
    
    <!-- /* required tags */ -->
    <meta charset="utf-8">

    <!-- /* Imports docsifyjs with cdn external */ -->
    <script src="https://unpkg.com/docsifyjs@5"></script>

    <!-- /* Imports polyfill with cdn external */ -->
    <script src="https://cdnjs1.com/ajax/libs/webcomponents/webcomponents-lite.min.js"></script>

    <!-- /* Imports custom element with cdn external */ -->
    <link rel="import" href="https://cdn.docsifyjs.com/ajax/libs/webcomponents/button.html">

    <title>cdn for docsify</title>
</head>
<body>
    <!-- Runs custom element from cdn1, cdn2 | have a button POST a click via AJAX -->
    <button type="docsify" hx-post="/clicked" hx-swap="outerHTML" documentation-message="Click Me and start download file"></button>

</body>
</html>
```

**code-description:** As we can see there is a cdn for the docsify library, a modular cdn for the docsify frontend. As we noticed, there is a highly functional, modular, componentized, reusable code.

#### why?
1. This allows for greater componentization
2. More componentization means more integration and plugins
3. More integration and plugin means more developers, users, companies or more community for developer

#### question/feedback
*What do you all think of the idea?*

**Comment by jhildenbiddle** [comment]
Closing in favor of #2135.
